### PR TITLE
Duplicate sequence number handling

### LINKED ISSUE #258: [Consumer] Do not accept the same sequence number twice
Also produce a warning message if the document content would otherwise be different from the one we already have in the sequence object.

### PR DESCRIPTION
**Pull request by kozmaz87**
Resolve #258 and address requirement R107.

### CODE REVIEW

File: ebu_tt_live/documents/ebutt3.py [hunk 1]
```diff
@@ -7,9 +7,10 @@
 from ebu_tt_live.strings import ERR_DOCUMENT_SEQUENCE_MISMATCH, \
     ERR_DOCUMENT_NOT_COMPATIBLE, ERR_DOCUMENT_NOT_PART_OF_SEQUENCE, \
     ERR_DOCUMENT_SEQUENCE_INCONSISTENCY, DOC_DISCARDED, DOC_TRIMMED, DOC_REQ_SEGMENT, DOC_SEQ_REQ_SEGMENT, \
-    DOC_INSERTED, DOC_SEMANTIC_VALIDATION_SUCCESSFUL, ERR_SEQUENCE_FROM_DOCUMENT
+    DOC_INSERTED, DOC_SEMANTIC_VALIDATION_SUCCESSFUL, ERR_SEQUENCE_FROM_DOCUMENT, \
+    ERR_DOCUMENT_SEQUENCE_SEQID_COLLISION
```
Comment (nigelmegitt) [inline ebu_tt_live/documents/ebutt3.py]: Shouldn't this be `...SEQNUM_COLLISION` ?

File: ebu_tt_live/strings.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 ERR_TIME_WRONG_FORMAT = gettext('Wrong time format. datetime.timedelta is expected')
 ERR_TIME_FORMAT_OVERFLOW = gettext('Time value is out of format range')
 ERR_DOCUMENT_SEQUENCE_MISMATCH = gettext('sequenceIdentifier mismatch')
+ERR_DOCUMENT_SEQUENCE_SEQID_COLLISION = gettext('Sequence identifier: [{sequence_identifier}] already present in sequence: [{sequence_identifier}]')
```
Comment (nigelmegitt) [inline ebu_tt_live/strings.py]: Shouldn't this be " `{sequence_number}` already present ... "?

Comment (kozmaz87) [inline ebu_tt_live/strings.py]: Oh my yes. I was lost there in typing. Correcting. Nice catch.

**Comment by nigelmegitt** [comment]
@EyalLavi please could you look at this and see if it is easy to make a BDD for it, or if the unit test is sufficient (I think it may be) then add an entry to `SPEC-CONFORMANCE.md` (in this branch)?

**Comment by nigelmegitt** [comment]
@kozmaz87 please could you look at pull request #336 and see what it would take to merge that into here and implement the BDD that's in there?

**Pull request merged**